packages/prisma/migrations/20250617123959_update_handle_routing_form_response_fields/migration.sql

```diff
@@ -0,0 +1,207 @@
+                            VALUES (
+                                response_id,
+                                field_record->>'id',
+                                ARRAY[response_field->>'value']
+                            );
+                        END IF;
+                    EXCEPTION WHEN OTHERS THEN
+                        RAISE WARNING 'Failed to insert multiselect values for field %', field_record->>'id';
+                    END;
+                END IF;
+            ELSIF field_type = 'number' THEN
+                -- Handle number values
+                -- Accept both JSON number values and string values that can be converted to numbers
+                IF response_field->>'value' IS NOT NULL THEN
+                    BEGIN
+                        IF jsonb_typeof(response_field->'value') = 'number' THEN
+                            -- If it's already a JSON number, use it directly
+                            INSERT INTO "RoutingFormResponseField" ("responseId", "fieldId", "valueNumber")
+                            VALUES (
+                                response_id,
+                                field_record->>'id',
+                                (response_field->>'value')::decimal
+                            );
+                        ELSIF jsonb_typeof(response_field->'value') = 'string' THEN
+                            -- If it's a string, try to convert it to a number
+                            INSERT INTO "RoutingFormResponseField" ("responseId", "fieldId", "valueNumber")
+                            VALUES (
+                                response_id,
+                                field_record->>'id',
+                                (response_field->>'value')::decimal
+                            );
+                        END IF;
+                    EXCEPTION WHEN OTHERS THEN
+                        RAISE WARNING 'Failed to insert number value for field % (value: %)', field_record->>'id', response_field->>'value';
+                    END;
+                END IF;
+            ELSIF field_type = 'select' THEN
+                -- Handle select values - can be either string or array
+                IF response_field->>'value' IS NOT NULL THEN
+                    BEGIN
+                        IF jsonb_typeof(response_field->'value') = 'array' THEN
+                            -- If it's an array, take the first element
+                            INSERT INTO "RoutingFormResponseField" ("responseId", "fieldId", "valueString")
+                            VALUES (
+                                response_id,
+                                field_record->>'id',
+                                (response_field->'value'->>0)
+                            );
+                        ELSIF jsonb_typeof(response_field->'value') = 'string' THEN
+                            -- If it's a string, use it directly
```

Comment:
Q: When will it be a string, and when will it be an array, and is taking the first element in case of an array sufficient? (e.g., don't we support like a multiselect in routing form where users can choose multiple options? I might be very wrong here, but just double checking)

Reply:
In case of 'select', it's supposed to be a string, but unfortunately we do have cases where it is an array of string with one item in it. This logic is to gracefully denormalize weird cases.

Also, 'multiselect', it's supposed to be an array of string, but there is a case where it's just a string. We convert it to an array of string as well.